.github/workflows/ci.yml

```diff
@@ -1,29 +1,123 @@
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [closed]
+    branches:
+      - main
+  release:
+    types: [published]
 
 jobs:
-  build:
+  build-and-test:
     runs-on: ubuntu-latest
+    if: github.event_name == 'push' || github.event.pull_request.merged == true
     strategy:
       matrix:
         python-version: ["3.7", "3.8", "3.9", "3.10", "3.11", "3.12"]
     steps:
       - uses: actions/checkout@v3
+
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v3
         with:
           python-version: ${{ matrix.python-version }}
+
       - name: Install dependencies
         run: |
           sudo apt-get update
-          sudo apt-get install libhdf5-serial-dev zlib1g-dev libatlas-base-dev lcov swig3.0 libpcsclite-dev
+          sudo apt-get install -y libhdf5-serial-dev zlib1g-dev libatlas-base-dev lcov swig3.0 libpcsclite-dev
           mkdir -p ~/bin/ && ln -s /usr/bin/swig3.0 ~/bin/swig && export PATH=~/bin/:$PATH
-          pip install -e .[ci,test,yubikey] # Install with CI deps
+          pip install -e .[ci,test,yubikey]
+
       - name: setup github user
-        shell: bash
         run: |
           git config --global user.name oll-bot
           git config --global user.email developers@openlawlib.org
+
       - name: Run pre-commit and test with pytest
         run: |
           pre-commit run --all-files
           pytest taf/tests
+
+  build-and-test-executables:
```

Comment:
We could run this only if the the previous job is successful. That way, we won't build the executables if tests fail for some reason. You could do that like this:
```
    needs: run-tests
    runs-on: ubuntu-latest
    if: ${{ needs.run-tests.result == 'success' }}
```
If the first job is named `run-tests`